### DISCUSSION TITLE
Float being ignored

### DISCUSSION THREAD

**Discussion by code-monki** [jgm/pandoc#10329] [Q&A]
Hi!

I'm somewhat new to pandoc and I'm having trouble figuring out how to properly get an HTML document to convert to a PDF document.

I've tried:

```
    pandoc -f html -t pdf -o test.pdf --standalone test.html
```

Pandoc produces a PDF but it seems to ignore the embedded CSS (I'm aware of the flag to include external CSS, but I'm trying to use as simple of an example as possible &mdash; see attached files).

The issue that I'm seeing is that if I display the HTML file in the browser, the image that is floated to the right appears correctly while in the PDF it appears the float CSS directive is ignored. My goal is to be able to embed images and have the text wrap around them, but I'm sort of stumped as to how to do that and I'm not really eager to have to load each file into the browser and print them out (which does, incidentally produce the correct output).

Any thoughts, pointers, spare change, etc., appreciated and thanks in advance for your help!

MacOS X Sonoma, 14.6.1
Pandoc v3.5

[test-files.zip](https://github.com/user-attachments/files/17494595/test-files.zip)

**Comment by jgm** [comment]
Correct. Pandoc pays no attention to CSS.  It is concerned with capturing document structure, not formatting details.

**Reply by code-monki** [reply]
So even if I'm including the CSS, pandoc will ignore the CSS?

**Reply by jgm** [reply]
Yes.

**Reply by code-monki** [reply]
Hmm, that makes it less useful for me then.

Thanks for your response.

**Reply by jdlail** [reply]
I did this to get an image to float left or right with a caption in pdf output. I don't know if it's the best way, but it produced the output I wanted.

~~~html
<figure style="float: right; padding: 0 0 0 0; margin: 0 0 0 0; width: 50%">

![image caption](path-to-image "image caption"){style="width: 105%;"}

</figure>
~~~
